### PR TITLE
46974426 - pull request for merging 2d UNet into PatternAnalysis-2024 topic recognition branch

### PR DESCRIPTION
**Pull request by robert123654**
Dear Shakes,

Please review my contribution to the PatternAnalysis-2024 repository and provide feedback if not appropriate for merging.

Thanks,

Robert

### CODE REVIEW

**Comment by yexincheng** [comment]
**This is an initial inspection, no action is required at this point**
1. **Solves problem**: The final DICE score on validaiton set is around 0.998 from [images/50Epoc_validation_score.png](https://github.com/robert123654/PatternAnalysis-2024/blob/topic-recognition/recognition/2d_unet_s46974426/images/50Epoc_validation_score.png). Training loss is provided. Prediction figures were not uploaded. Figures were not inserted into readme. Dependency and reference are not written. The report is a bit informal and hard to find the important information.
2. **Implementation**: Well-implemented functions and classes. Used ArgParse to set hyerparameters.
3. **Design**: Great
4. **Comment**: Sufficient
5. **Difficulty**: Easy

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Easy. UNet2D (Easy Difficulty)</td><td>-10</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. Missing plots/figures. Report tone incorrect. </td><td>-4</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Description and Comments :Good. No references.</td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). Submitted on time.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. </td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-18</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Robert Slomka

**Comment by shakes76** [comment]
Likely data leakage with such high validation score -1
